qiskit_machine_learning/algorithms/effective_dimension.py

```diff
@@ -0,0 +1,355 @@
+        """Returns the dimension of the model according to the definition
+        from [1]."""
+        return self._model.num_weights
+
+    @property
+    def params(self) -> np.ndarray:
+        """Returns network parameters."""
+        return self._params
+
+    @params.setter
+    def params(self, params: Optional[Union[List[float], np.ndarray, float]]) -> None:
+        """Sets network parameters."""
+        if params is not None:
+            self._params = np.asarray(params)
+            self._num_params = len(self._params)
+        else:
+            # random sampling from uniform distribution
+            self._params = algorithm_globals.random.uniform(
+                0, 1, size=(self._num_params, self._model.num_weights)
+            )
+
+    @property
+    def inputs(self) -> np.ndarray:
+        """Returns network inputs."""
+        return self._inputs
+
+    @inputs.setter
+    def inputs(self, inputs: Optional[Union[List[float], np.ndarray, float]]) -> None:
+        """Sets network inputs."""
+        if inputs is not None:
+            self._inputs = np.asarray(inputs)
+            self._num_inputs = len(self._inputs)
+        else:
+            # random sampling from normal distribution
+            self._inputs = algorithm_globals.random.normal(
+                0, 1, size=(self._num_inputs, self._model.num_inputs)
+            )
+
+    def run_monte_carlo(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        This method computes the model's Monte Carlo sampling for a set of
+        inputs and parameters (params).
+
+        Returns:
+             grads: QNN gradient vector, result of backward passes, of shape
+                ``(num_inputs * num_params, output_size, num_weights)``
+             outputs: QNN output vector, result of forward passes, of shape
+                ``(num_inputs * num_params, output_size)``
+        """
+        grads = np.zeros(
+            (
+                self._num_inputs * self._num_params,
+                self._model.output_shape[0],
+                self._model.num_weights,
+            )
+        )
+        outputs = np.zeros((self._num_inputs * self._num_params, self._model.output_shape[0]))
+
+        for (i, param_set) in enumerate(self.params):
+            t_before_forward = time.time()
+            fwd_pass = np.asarray(self._model.forward(input_data=self.inputs, weights=param_set))
+            t_after_forward = time.time()
+
+            if self._callback is not None:
+                msg = f"iteration {i}, time forward pass: {t_after_forward - t_before_forward}"
+                self._callback(msg)
+
+            back_pass = np.asarray(
+                self._model.backward(input_data=self.inputs, weights=param_set)[1]
+            )
+            t_after_backward = time.time()
+
+            if self._callback is not None:
+                msg = f"iteration {i}, time backward pass: {t_after_backward - t_after_forward}"
+                self._callback(msg)
```

Comment:
I wonder why you decided to pass a message string instead of numbers to the callback?